File: 22-Apr-Java-3.30-Karen/Classwork/LectureFlow/ConstructorDemo.java
/*
 * Constructor :- Is a special member function whose task is to initialize
 * 				  object of its class.
 * 				  
 * Features : 	They have same name as class name.
 * 				No Return Types
 * 				Invocation is done automatically on creation of the object.
 * 
 * Types of Constructor:
 * 
 * 1) Default Constructor
 * 2) Parameterized Constructor
 * 3) Copy Constructor : it always takes object as arguments.
 * 
 * 
 * Parameterized and Copy Constructor are called as Constructor Overloading.
 * 
 * 				
 * 
 * 
 */


class Box{
	double width, height, depth;
	
	Box(){
		System.out.println("\nDefault Constructor Called");
		width = 4;
		height = 5;
		depth = 3;
	}
	
	Box(double w, double h, double d){
		System.out.println("\nParameterized Constructor Called");
		width = w;
		height = h;
		depth = d;
	}
	
	
	Box(Box b){
		System.out.println("\nCopy Constructor Called");
		width = b.width;
		height = b.height;
		depth = b.depth;
	}
	
	
	
	void volume() {
		System.out.println("Volume of Box is "+ (width*height*depth));
	}
}



public class ConstructorDemo {
	public static void main(String[] args) {
		
		Box b1 = new Box();
		b1.volume();
		Box b2 = new Box(14,12,3);
		b2.volume();
		Box b3 = new Box(b1);
		b3.volume();
	}

}

File: 5-Apr-C-C++/Assignments/C Assignments/Dhruvam/constructordemo.cpp
#include<iostream>
using namespace std;
class box
{
	public:
			double width,height,depth;
			box()
			{
				width=5;
				height=5;
				depth=4;
				cout<<"default constructor called"<<endl;
			}
			box(double w, double h, double d)
			{
				cout<<"\n\n parameterized constructor called."<<endl;
				width=w;
				height=h;
				depth=d;
			}
			box(const box & chandu)
			{
				cout<<"\n \n copy constructor called."<<endl;
				width=chandu.width;
				height=chandu.height;
				depth=chandu.depth;
			}
			void volume()
			{
				cout<<"volume of box:"<<(width*height*depth);
			}
};

int main()
{
	box b;
	b.volume();
	
	box b1(10,3,7);
	b1.volume();
	
	box b2(b1);
	b2.volume();
	
	box*ptr=new box();
	ptr->volume();
	
	box *ptr1=new box(4,7,9);
	ptr->volume();
	
	box*ptr2=new box(*ptr1);
	ptr2->volume();
}